### ISSUE TITLE
False positive for `too-many-arguments` in methods

### ISSUE THREAD

**Issue by mbyrnepr2** [PyCQA/pylint#8675]
### Bug description

As mentioned in [merge-request 8674 comment](https://github.com/pylint-dev/pylint/pull/8674#issuecomment-1543452820), `self` is counted as one of the arguments in the `too-many-arguments` calculation.
This means that if a user creates a function with 5 parameters, the message is _not_ emitted whereas if the user creates a method with the same 5 parameters, then the message _will_ be emitted. This behaviour could be surprising.

No message emitted:
```python
def name1(param1, param2, param3, param4, param5):
    ...
```

Message is emitted:
```python
class A:
    def name2(self, param1, param2, param3, param4, param5):  # R0913: Too many arguments (6/5) (too-many-arguments)
        ...
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
R0913: Too many arguments (6/5) (too-many-arguments)
```

### Expected behavior

No message emitted for this example

### Pylint version

```shell
pylint 3.0.0b1
astroid 3.0.0a2
Python 3.10.4
```

**Comment by jacobtylerwalls** [comment]
I imagine this will be tricky when one but not the other is a staticmethod.
